docs/platform_developers/database.rst

```diff
@@ -2,37 +2,40 @@
 Database Schema Management
 **************************
 
-This page describes how database schema updates are managed by the Inmanta core.
+In some situation a change to the database schema is required. To perform these database schema
+migration, we implemented a migration tool and associated testing framework. This page describes how to create a new
+version of the database schema and test the migration script.
 
 
 Definition new schema version
 #############################
 
+The version number of the database schema evolves independently from any other versioned Inmanta element
+(product version, extension version, etc.). Each commit can introduce changes to the database schema. When that happens
+the commit creates a new database schema version. This means that multiple schema version can exist between two
+consecutive stable releases of the orchestrator.
+
 A new version of the database schema is defined by adding a new Python module to the ``inmanta.db.versions`` package. The
-name of this module should have the format ``v<version>.py``, where <version> is an integer indicating the version of
-the new database schema. Version numbers start at 1.
+name of this module should have the format ``v<timestamp><i>.py``, where the timestamp is in the form
+``YYYYMMDD`` and ``i`` is an index to allow more than one schema update per day (e.g. ``v202102220.py``).
 
 Each of these Python modules should implement an asynchronous function ``update`` that accepts a database connection object
 as an argument. This function should execute all database queries required to update from the previous version of the
-database schema (<version> - 1) to the new version of the database schema (<version>). **All changes done by the update
-function should be executed in the transaction.** An example is given in the code snippet below.
+database schema to the new version of the database schema.
 
-Each each of these Python modules must also contain the field ``DISABLED`` set to false to make the changes effective.
+An example is given in the code snippet below:
 
 .. code-block:: python
 
-    # File: src/inmanta/db/versions/v1.py
+    # File: src/inmanta/db/versions/v202102220.py
     from asyncpg import Connection
 
-    DISABLED = False
-
     async def update(connection: Connection) -> None:
         schema = """
         ALTER TABLE public.test
         ADD COLUMN new_column;
         """
-        async with connection.transaction():
-            await connection.execute(schema)
+        await connection.execute(schema)
```

Comment:
You removed this because all updates are already in a transaction? As in, one transaction for all of them, not one for each of them.

Reply:
Yes. I removed it because all updates are already in a transaction. I don't think starting a sub-transaction adds any value here.

Reply:
Indeed, forgot to mark this as resolved after our discussion.